src/textual/widgets/__init__.py

```diff
@@ -1,21 +1,41 @@
-from ._footer import Footer
-from ._header import Header
-from ._button import Button
-from ._placeholder import Placeholder
-from ._static import Static
-from ._tree_control import TreeControl, TreeClick, TreeNode, NodeID
-from ._directory_tree import DirectoryTree, FileClick
+from __future__ import annotations
+from importlib import import_module
+import typing
 
+from ..case import camel_to_snake
+
+if typing.TYPE_CHECKING:
+    from ..widget import Widget
+
+
+_WIDGETS_LAZY_LOADING_CACHE: dict[str, type[Widget]] = {}
```

Comment:
We might not need this. I suspect the import machinery will cache for us.

Could you move `__all__` to the top of the file, just to make it clear what we need to update.

Reply:
After having checked, it seems that each module importing a given widget will activate the `__get__` method - after that it's kept in the caller module's cache, but if 3 modules do `from textual.widgets import Placeholder` it will still be called 3 times.
So we'd better keep that cache. I moved the `__all__` to the top of the file, though  :slightly_smiling_face: [0bdd2be](https://github.com/Textualize/textual/pull/584/commits/0bdd2be141c6227af26bdb4d30bdce193ae4623b)